File: 03_Two_Pointers/01_Valid_Palindrome/125_Valid_Palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s =  ''.join(ch for ch in s if ch.isalnum()).lower()

        mid = int(len(s) / 2)
        i = 0
        while i != mid:
            if (s[i] != s[len(s) - 1 - i]):
                return False
            i +=1

        return True

File: 03_Two_Pointers/01_Valid_Palindrome/125_Valid_Palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {

        string cleaned_s;

        for (char c: s) {
            if (isalnum(c)){
                cleaned_s += tolower(c);
            }
        }

        int mid = cleaned_s.size() / 2;
        
        int i = 0;
        while (i < mid){
            if (cleaned_s[i] != cleaned_s[cleaned_s.size() - 1 - i]){
                return false;
            }
            i++;
        }

        return true;
        
    }
};
